src/metricscollector/server/memory_metric_handler.go

```diff
@@ -67,3 +71,75 @@ func (h *MemoryMetricHandler) GetMemoryMetric(w http.ResponseWriter, r *http.Req
 
 	w.Write(body)
 }
+
+func (h *MemoryMetricHandler) GetMemoryMetricHistory(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appId := vars["appid"]
+	startParam := r.URL.Query()["start"]
+	endParam := r.URL.Query()["end"]
+	h.logger.Debug("get-memeory-metric-history", lager.Data{"appId": appId, "start": startParam, "end": endParam})
+
+	var err error
+	start := int64(0)
+	end := int64(-1)
+
+	if startParam != nil {
```

Comment:
still need to check whether "startParam == nil" or "len(startParam) == 0". In this case, start will remain the default value 0. 
Anyway changed to use len() for all cases